### ISSUE TITLE
[frameit] When using the stacked option and a single-line keyword and multi-line title, there is a large vertical padding around the keyword

### ISSUE THREAD

**Issue by funnel20** [fastlane/fastlane#11646]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When defining a single-line keyword and multi-line title and enable option `stack_title`, there is a large vertical padding around the keyword:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35216577-5abb00f2-ff68-11e7-8fec-08dfc7ed1324.png)

When changing the title text to single-line; the expected vertical padding between keyword and title is applied:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35216626-851cb03e-ff68-11e7-951c-4c33b6d96d83.png)

This issue is discovered during the analysis of issue #11585, see https://github.com/fastlane/fastlane/issues/11585#issuecomment-358717961.

**Cause**
This large padding is introduced in my solution for issue #10440, to align the text images when the keyword and title are used side by side.

##### Complete output when running fastlane, including the stack trace and command used

[INSERT OUTPUT HERE]

### Environment

Fastlane 2.76.1

**Comment by funnel20** [comment]
**Solution**
The enhanced trim box adjustments to vertically align the keyword and title should not occur when option `stack_title` is enabled.

**Comment by janpio** [comment]
Oh right, of course, your trim box adjustments only make sense if they are shown next to each other. 👍

**Comment by funnel20** [comment]
To indicate the issue I will temporarily use the yellow background again by adding these 2 lines to the existing `do` (see: https://github.com/fastlane/fastlane/pull/11096#issuecomment-348940286):
```Ruby
title_image.combine_options do |i|
  # Oversize as the text might be larger than the actual image. We're trimming afterwards anyway
  i.resize "#{max_width * 5.0}x#{image_height}!" # `!` says it should ignore the ratio
  i.background 'yellow'
  i.alpha 'remove'
end
```

**Comment by funnel20** [comment]
**Analysis**
The effect of the fix for issue #10440 when using keyword and title side-by-side (e.g. `stack_title` is false) is that one or both trim boxes are made taller to have the same text baseline for non multi-line text. In this example the title trim box is made taller at the bottom:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35274776-b324b91a-003d-11e8-882d-fece0a65152d.png)

When using a multi-line title, the keyword is made taller so it's centred (more or less)vertically:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35274750-97b1175a-003d-11e8-80b4-97e53f31168b.png)

When enabling `stack_title` for the same text, the keyword trim box is still enlarged to match that of the title:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35274894-29115250-003e-11e8-89eb-fd309e1a9ca7.png)

**This is exactly the issue.**

**Comment by funnel20** [comment]
**Solution**

- Add `stack_title`  as 3rd input argument to method `build_text_images()`
- When `stack_title` is true, the trim box adjustment is omitted in the `words.each` loop to crop the image.
- Updated comments to reflect these changes.

Now the trim box for the title is not enlarged anymore:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35274885-1e06e230-003e-11e8-93e8-8bd54b09539f.png)

**Comment by janpio** [comment]
Looking good.

What defines the space between "Keyword" and "This is ..."?

**Comment by funnel20** [comment]
But wait...
Why is the vertical distance between the keyword and the multi-line title larger than with a keyword and a single-line title???

![35274885-1e06e230-003e-11e8-93e8-8bd54b09539f](https://user-images.githubusercontent.com/2139844/35275321-a5e76e8a-003f-11e8-9e38-95cc256c5edf.png)

![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35275398-e8e185b8-003f-11e8-8ba7-f6a510d2bf4b.png)

**Comment by funnel20** [comment]
A better variable to define the spacing is actual_font_size, as this is independent of the number of lines.
When using:
`spacing_between_title_and_keyword = (actual_font_size / 2)`

The result will be 2 equal distances:
![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35275799-2ac6c190-0041-11e8-8656-1778f997c248.png)

![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35275762-1486d302-0041-11e8-9ce5-9d39e2e98f6e.png)

**Comment by funnel20** [comment]
If this all looks plausible I happy to create a new PR.

**Comment by janpio** [comment]
Looks perfect. Go for it!

(Also works for 2 line keyword but one line title?) 
("keyword" and "title" are suboptimal names for these things...)

**Comment by funnel20** [comment]
Yes sir:

![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35276288-34079f52-0043-11e8-910e-a2437762a1cb.png)

![iphone se-01_remotetab_framed](https://user-images.githubusercontent.com/2139844/35276358-6ce13f9a-0043-11e8-8c7a-f8d0ebbef8af.png)

**Comment by funnel20** [comment]
This issue can be closed as PR #11661 has been released in version 2.79.0.
